Вопросы для самоконтроля:
1.Правильно. Блок finally будет выполняться в любом случае, независимо от catch, например, будем закрывать ресурсы.
2.Обработчик перехватывает все исключения, которые являются его подклассами. Такой обработчик будет перехватывать
все исключения, а пользователю будет непонятно, какое именно исключение вышло(Error, Runtime)
3.Так как catch (Exception e)  catch (ArithmeticException a) не будет иметь смысла, так как все исключения перехватит первый.
Компилятор не позволит нам так сделать, поскольку более специфичные исключения должны обрабатываться перед более общими.
4. Сопоставьте каждую ситуацию из первого списка с пунктом второго списка:
 int[] A; A[0] = 0 - Ошибка компиляции.
 JVM начинает запускать вашу программу, но обнаруживает, что не может найти классы Java платформы - Ошибка
 Программа читает входной поток и достигает маркера конца потока - Исключение не произойдёт
 После достижения маркера конца входного потока, но до закрытия самого потока, программа пытается считать данные - Проверяемое исключение